Fix/groupname mock user data

jipatino98 opened the pull request:
### Description

This PR introduces some minor changes including:
- Group Name appears as the Main Header
- Mock user data is replace with actual user data from database

NOTE: When a user has no assign role returned, the default output is "No Role Assigned"

### Issue Link

https://cascarita.atlassian.net/browse/CV-113

### Screenshots 

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/810cc17b-bd07-410f-8b88-6582ad7820b9" />

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/d313bb39-6ac2-48e0-a8c5-3ac8a53362c4" />


### Checklist

- [ ] I have tested the changes locally by running `npm run test`
- [ ] I have added or updated relevant documentation
- [ ] I have autoformatted the code by running `npm run eslint`
- [ ] I have added test cases (if applicable)

### Additional Notes

abanuelo reviewed (approved):
Address the nit, but all else looks great!

jipatino98 commented on client/src/pages/Users/types.ts:
```diff
@@ -1,11 +1,13 @@
+  phone_number: string;
+  internally_created: boolean;
+  picture: string;
+  UserRoles: UserRole[]; // Enforces only allowed values
```
You got it! updated to reflect that

jipatino98 commented:
Updated `User` types so that all types are declared in snake_case, addressed by @abanuelo. I also linked user's online account to the link icon in Stripe Account page.
